### PR TITLE
MW-1159: AddProductPage for issue - generic approach, fixed reducers

### PR DESCRIPTION
**Pull request by sniedzielski**
TICKET; https://openlmis.atlassian.net/browse/MW-1159

- add product page for Issue. 
- additional changes in reducers to avoid conflicts with 'Store' while restarting/reloading app.

### CODE REVIEW

File: src/stock-adjustment-mobile/adjustment-form.component.jsx [hunk 1]
```diff
@@ -22,20 +22,19 @@ import InlineField from '../react-components/form-fields/inline-field';
...
 
     const dispatch = useDispatch();
-    const adjustment = useSelector(state => state.adjustment.adjustment);
-    const userHomeFacility = useSelector(state => state.facilities.userHomeFacility);
-    const program = useSelector(state => state.program.program);
-    const toastList = useSelector(state => state.toasts.toasts);
+    const adjustment = useSelector(state => state[`adjustment${adjustmentType}`][`adjustment${adjustmentType}`]);
+    const userHomeFacility = useSelector(state => state[`facilities${adjustmentType}`][`userHomeFacility${adjustmentType}`]);
+    const program = useSelector(state => state[`program${adjustmentType}`][`program${adjustmentType}`]);
+    const toastList = useSelector(state => state[`toasts${adjustmentType}`][`toasts${adjustmentType}`]);
```
Comment (rstencel) [inline src/stock-adjustment-mobile/adjustment-form.component.jsx:37]: Code readability is horrible here. Maybe you could use 
```
const adjustment = useSelector(state => adjustmentType === ISSUE ? state.adjustment... : state.adjustment...)
```
Also I would recommend to create constant values for `adjustmentType`s: ISSUE and ADJUSTMENT. Image one typo which causes the app to break!

File: src/stock-adjustment-mobile/program-select.jsx [hunk 2]
```diff
@@ -49,21 +49,45 @@ const ProgramSelect = ({ offlineService, stockReasonsFactory, existingStockOrder
...
+                    });
+                }   
             });
         }
     };
 
+    const goToProductPage = (programId, programObject, program) => {
+        sourceDestinationService.getDestinationAssignments(
+            programId, facility.id
+        ).then(sourceDestinations => {
```
Comment (rstencel) [inline src/stock-adjustment-mobile/program-select.jsx:78]: Formatting looks bad. I think
```
sourceDestinationService.getDestinationAssignments(programId, facility.id).then(sourceDestinations => {
```
is just fine

**Review by rstencel — CHANGES_REQUESTED** [review]
A few constants should be created. Also, code readability should be improved.

File: src/stock-issue-mobile/issue-app.jsx [hunk 3]
```diff
@@ -50,6 +59,15 @@ const IssueApp = ({
                 hashType="hashbang"
             >
                 <Switch>
+                    <Route path="/makeIssueAddProducts">
+                        {   
+                            userHomeFacility
+                            && <AddProductsPage
+                                adjustmentType="Issue"
```
[the remark this answers is not in the document]
Comment (sniedzielski) [reply src/stock-issue-mobile/issue-app.jsx]: fixed

File: src/stock-issue-mobile/reducers/adjustment.jsx [hunk 4]
```diff
@@ -15,26 +15,26 @@
...
         }
     }
 });
 
-export const {setAdjustment} = adjustmentSlice.actions;
-export const {appendToAdjustment} = adjustmentSlice.actions;
-export const {resetAdjustment} = adjustmentSlice.actions;
+export const {setAdjustment} = adjustmentIssueSlice.actions;
+export const {appendToAdjustment} = adjustmentIssueSlice.actions;
+export const {resetAdjustment} = adjustmentIssueSlice.actions;
```
[the remark this answers is not in the document]
Comment (sniedzielski) [reply src/stock-issue-mobile/reducers/adjustment.jsx]: fixed

File: src/stock-adjustment-mobile/reducers/program.jsx [hunk 5]
```diff
@@ -15,22 +15,22 @@
...
-            state.program = null;
+            state.programAdjustment = null;
         }
     }
 });
 
-export const {setProgram} = programSlice.actions;
-export const {resetProgram} = programSlice.actions;
+export const {setProgram} = programAdjustmentSlice.actions;
+export const {resetProgram} = programAdjustmentSlice.actions;
```
[the remark this answers is not in the document]
Comment (sniedzielski) [reply src/stock-adjustment-mobile/reducers/program.jsx]: fixed

File: src/stock-adjustment-mobile/reducers/adjustment.jsx [hunk 6]
```diff
@@ -15,26 +15,26 @@
...
         }
     }
 });
 
-export const {setAdjustment} = adjustmentSlice.actions;
-export const {appendToAdjustment} = adjustmentSlice.actions;
-export const {resetAdjustment} = adjustmentSlice.actions;
+export const {setAdjustment} = adjustmentAdjustmentSlice.actions;
+export const {appendToAdjustment} = adjustmentAdjustmentSlice.actions;
+export const {resetAdjustment} = adjustmentAdjustmentSlice.actions;
```
[the remark this answers is not in the document]
Comment (sniedzielski) [reply src/stock-adjustment-mobile/reducers/adjustment.jsx]: fixed

File: src/stock-adjustment-mobile/program-select.jsx [hunk 7]
```diff
@@ -49,21 +49,45 @@ const ProgramSelect = ({ offlineService, stockReasonsFactory, existingStockOrder
...
-                    dispatch(setReasons(mappedReasons));
-                    return mappedReasons
-                }).then(mappedReasons => {
-                    if (programSelected.programId !== program.id) {
-                        dispatch(resetAdjustment(adjustment));
-                    }
-                    dispatch(setProgram(programObject));
-                    history.push(`/make${adjustmentType}AddProducts`);
-                });   
+                let reasonType = adjustmentType === "Adjustment" ? null : (adjustmentType === "Issue" ? "DEBIT" : "CREDIT" );
```
[the remark this answers is not in the document]
Comment (sniedzielski) [reply src/stock-adjustment-mobile/program-select.jsx]: fixed

File: src/stock-adjustment-mobile/program-select.jsx (hunk 2 above)
Comment (sniedzielski) [reply src/stock-adjustment-mobile/program-select.jsx]: fixed

**Comment by sniedzielski** [comment]
@rstencel I applied some fixes here https://github.com/OpenLMIS/openlmis-stockmanagement-ui/pull/23/commits/67ce4ab9a7cc026538468a91edbcccde91a35c71

**Review by rstencel — APPROVED** [review]
In the future a separate file with all constants should be created. Apart from that these changes LGTM.

**Pull request merged**